packages/wagmi/src/index.ts

```diff
@@ -23,11 +28,18 @@ export function bindings(): IBindings {
         }
       }
 
-      const signer = await fetchSigner<RequiredSigner>({ chainId });
-
-      invariant(signer, 'Cannot get signer, is the wallet connected?');
+      const publicClient = getPublicClient<PublicClient>({ chainId });
 
-      return signer;
+      return ethersProviderFromPublicClient(publicClient).getSigner();
     },
   };
 }
+
+export function ethersProviderFromPublicClient(publicClient: PublicClient): providers.Web3Provider {
+  invariant(window.ethereum, 'window.ethereum is not defined');
+
+  return new providers.Web3Provider(
+    window.ethereum as providers.ExternalProvider,
```

Comment:
is this actually using anything from `viem`? is it safe to depend on `window.ethereum` or viem is setting that up?

Reply:
I also see sth like this in the discussion here https://github.com/wagmi-dev/wagmi/discussions/2340#discussioncomment-5935358
(instead of using `window.ethereum`)

```
const { connector } = useAccount()

// get the connected provider instead of window.ethereum
connector.getProvider().then((provider) => {
  return new Web3Provider(provider).getSigner()
})
```

Reply:
update this to use `walletClient.transport` which removes dependency on `window.ethereum` 👍